### PR TITLE
Equativ Bid Adapter: support native bid requests

### PR DESCRIPTION
**Pull request by jefftmahoney**
## Type of change
- [x] Feature

## Description of change
This work involves adding support for native requests to the new Equativ bid adapter.

Since this code base utilizes the ortbConverter library, there is little in the way of additional business logic updates in the adapter itself (beyond some warning logs). Most changes instead focus on unit tests verifying that bid requests sent out by the adapter conform to the Equativ SSP exchange's expectations of what should be present in the payloads.

### CODE REVIEW

File: modules/equativBidAdapter.js [hunk 1]
```diff
@@ -18,22 +19,22 @@ const LOG_PREFIX = 'Equativ:';
 const PID_COOKIE_NAME = 'eqt_pid';
 
 /**
- * Evaluates a bid request for validity. Returns false if the
- * request contains a video media type with no properties, true
- * otherwise.
+ * Evaluates impressions for validity.  The entry evaluated is considered valid if NEITHER of these conditions are met:
+ * 1) it has a `video` property defined for `mediaTypes.video` which is an empty object
+ * 2) it has a `native` property defined for `mediatTypes.native` which is an empty object
```
Comment (patmmccann) [inline modules/equativBidAdapter.js:24]: spelling

File: modules/equativBidAdapter.js [hunk 2]
```diff
@@ -151,6 +151,17 @@ export const converter = ortbConverter({
       });
     }
 
+    // "assets" is not included as a property to check here because the
+    // ortbConverter library checks for it already and will skip processing
+    // the request if it is missing
+    if (deepAccess(bid, 'mediaTypes.native')) {
+      ['privacy', 'plcmttype', 'eventtrackers'].forEach(prop => {
+        if (!bid.mediaTypes.native.ortb[prop]) {
+          logWarn(`${LOG_PREFIX} Property "${prop}" is missing from request`, bid);
```
Comment (patmmccann) [inline modules/equativBidAdapter.js:160]: can you add detail, eg ${LOG_PREFIX} Property "${prop}" is missing from request, ${LOG_PREFIX} will not bid

**Review by patmmccann — CHANGES_REQUESTED** [review]
nitpicks

File: modules/equativBidAdapter.js (hunk 2 above)
Comment (jefftmahoney) [reply modules/equativBidAdapter.js:160]: Our product managers want the warnings in there, but do want the bid request to be stopped if these particular properties are missing.  

If `assets` is missing, they are OK with the bid getting killed, but that is handled already by `native.js`.

If not bidding if these other properties are missing is a hard requirement, I can share that back to them.  Or if you would like the message further refined in another way (e.g. adding "it is strongly recommended you add ${prop}", etc.) we can do that too.

Comment (patmmccann) [reply modules/equativBidAdapter.js:160]: putting myself in the shoes of someone reading this message, i dont know what the outcome is. Tell the publisher the consequence.

File: modules/equativBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/equativBidAdapter.js:24]: here: `mediatTypes.native`

Comment (jefftmahoney) [reply modules/equativBidAdapter.js:24]: Just spotted it - thanks!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**